src/main/java/net/dv8tion/jda/api/exceptions/ErrorHandler.java

```diff
@@ -0,0 +1,418 @@
+     *
+     * @param  <T>
+     *         The type
+     *
+     * @return This ErrorHandler with the applied handler
+     */
+    @Nonnull
+    public <T> ErrorHandler handle(@Nonnull Class<T> clazz, @Nonnull Consumer<? super T> handler)
+    {
+        Checks.notNull(clazz, "Class");
+        Checks.notNull(handler, "Handler");
+        return handle(clazz::isInstance, (ex) -> handler.accept(clazz.cast(ex)));
+    }
+
+    /**
+     * Handle specific throwable types.
+     * <br>This will apply the specified handler if the throwable is of the specified type. The check is done using {@link Class#isInstance(Object)}.
+     *
+     * <h2>Example</h2>
+     * <pre>{@code
+     * public static void logErrorResponse(RestAction<?> action) {
+     *     action.queue(null, new ErrorHandler()
+     *         .handle(ErrorResponseException.class,
+     *             ErrorResponseException::isServerError,
+     *             (ex) -> System.out.println(ex.getErrorCode() + ": " + ex.getMeaning())));
+     * }
+     * }</pre>
+     *
+     * @param  clazz
+     *         The throwable type
+     * @param  condition
+     *         Additional condition that must apply to use this handler
+     * @param  handler
+     *         The alternative handler
+     *
+     * @param  <T>
+     *         The type
+     *
+     * @return This ErrorHandler with the applied handler
+     */
+    @Nonnull
+    public <T> ErrorHandler handle(@Nonnull Class<T> clazz, @Nonnull Predicate<? super T> condition, @Nonnull Consumer<? super T> handler)
+    {
+        Checks.notNull(clazz, "Class");
+        Checks.notNull(handler, "Handler");
+        return handle(
+            (it) -> clazz.isInstance(it) && condition.test(clazz.cast(it)),
+            (ex) -> handler.accept(clazz.cast(ex)));
+    }
+
+    /**
+     * Handle specific throwable types.
+     * <br>This will apply the specified handler if the throwable is of the specified type. The check is done using {@link Class#isInstance(Object)}.
+     *
+     * <h2>Example</h2>
+     * <pre>{@code
+     * public static void logErrorResponse(RestAction<?> action) {
+     *     action.queue(null, new ErrorHandler()
+     *         .handle(Arrays.asList(Throwable.class),
+     *             (ex) -> ex instanceof Error,
+     *             (ex) -> ex.printStackTrace()));
+     * }
+     * }</pre>
+     *
+     * @param  clazz
+     *         The throwable types
+     * @param  condition
+     *         Additional condition that must apply to use this handler, or null to apply no additional condition
+     * @param  handler
+     *         The alternative handler
+     *
+     * @return This ErrorHandler with the applied handler
+     */
+    @Nonnull
+    public ErrorHandler handle(@Nonnull Collection<Class<?>> clazz, @Nullable Predicate<? super Throwable> condition, @Nonnull Consumer<? super Throwable> handler)
+    {
+        Checks.notNull(clazz, "Class");
+        Checks.notNull(handler, "Handler");
```

Comment:
We need a `Checks.nonNull` here on `clazz`. Might be good to rename to `clazzes` to clarify that this is a collection.